app/models/concerns/traceable.rb

```diff
@@ -0,0 +1,15 @@
+require 'active_support/concern'
+
+module Concerns::Traceable
+  extend ActiveSupport::Concern
+
+  included do
+    def time(process:)
+      started = Time.now
+      logger.info "Process: '#{process}' started at #{started}"
+      yield
+      ended = Time.now
+      logger.info "Process: '#{process}' ended at #{ended}, duration: #{ended - started}"
```

Comment:
It would be good to log when an exception is raised.
Just capture the exception, trace it, and re-raise it